src/lavinmq/auth/authenticators/rabbit_http_auth.cr

```diff
@@ -0,0 +1,41 @@
+require "../authenticator"
+require "../../server"
+
+module LavinMQ
+  module Auth
+    class RabbitBackendAuthenticator < Authenticator
+      def initialize(@users : UserStore)
+      end
+
+      def authenticate(username : String, password : Bytes) : User?
+        user = @users[username]
```

Comment:
I don’t think we should require a user to be pre-registered in LavinMQ just to authorize. Instead, if they can authenticate via the HTTP backend, we can the go on to create a "temporary/remote" user with some permissions that are sent in with the connect packet.
This would be a new user class, and needs to be added before we can move forward with implementing "remote" authenticators. (OAuth2 will also need it)

Reply:
The authenticate method is defined to return a `User`. I think this is a good thing. To add the notion of `RemoteUser` and `LocalUser`, I think I'll make an abstract `User` class, transform the current `User` into `LocalUser`.
Is this what you had in mind?